src/solvers/refinement/string_refinement.cpp

```diff
@@ -1685,14 +1701,26 @@ exprt string_refinementt::get(const exprt &expr) const
     if(it!=found_length.end())
       return get_array(ecopy, it->second);
   }
+  else if(refined_string_typet::is_refined_string_type(ecopy.type()) &&
```

Comment:
@allredj and I discovered that the branch above does not actually get refined strings (and neither does the call to `supert`), and I couldn't figure out for sure how to besides this. What's a better way to do this?